File: T4/codigos2122/ProyectoModelos/app/src/main/java/com/example/proyectomodelos/adapters/AdaptadorMarcas.java
package com.example.proyectomodelos.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.proyectomodelos.R;
import com.example.proyectomodelos.utils.Marca;

import java.util.ArrayList;
import java.util.List;

public class AdaptadorMarcas extends BaseAdapter {

    private List<Marca> listMarcas;
    private Context context;

    public AdaptadorMarcas(List listMarcas, Context context) {
        this.listMarcas = listMarcas;
        this.context = context;
    }

    @Override
    public int getCount() {
        return listMarcas.size();
    }

    @Override
    public Object getItem(int position) {
        return listMarcas.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        Marca marcaActual = listMarcas.get(position);

        convertView = LayoutInflater.from(context).inflate(R.layout.fila_spinner_marca, parent, false);
        TextView textoMarca = convertView.findViewById(R.id.texto_marca_fila);
        ImageView imagenMarca = convertView.findViewById(R.id.imagen_marca_fila);

        textoMarca.setText(marcaActual.getMarca());
        imagenMarca.setImageResource(marcaActual.getImagen());

        return convertView;
    }
}

File: Kotlin/T3/codigos_ret/Concesionario/app/src/main/java/com/example/concesionario/adapter/AdaptadorMarcas.kt
package com.example.concesionario.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.SpinnerAdapter
import android.widget.TextView
import com.example.concesionario.R
import com.example.concesionario.model.Marca

class AdaptadorMarcas(var listaMarcas: ArrayList<Marca>, var context: Context) : BaseAdapter() {
    // cuantas filas tiene que renderizar
    override fun getCount(): Int {
        return listaMarcas.size
    }

    // dame el elemento de la fila
    override fun getItem(position: Int): Marca {
        return listaMarcas[position]
    }

    // dame el id del elemento que esta en la fila
    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    // como se representa cada fila
    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        // el XML Para poder poder pintarlo
        // int -> vista
        val view: View = LayoutInflater.from(context)
            .inflate(R.layout.item_marca, parent, false)
        // los elementos que vas a personalizar
        val imagen: ImageView = view.findViewById(R.id.imagenSpinnerMarca);
        val texto: TextView = view.findViewById(R.id.textoSpinnerMarca);

        // poner cada cosa en su sitio Imagen de la marca -> imagen Nombre de la marca -> texto
        val marcaPosicion = listaMarcas[position]
        texto.text = marcaPosicion.nombre
        imagen.setImageResource(marcaPosicion.imagen!!)
        return view
    }
}